op-program/scripts/run-static-analysis.sh

```diff
@@ -0,0 +1,117 @@
+#!/usr/bin/env bash
+
+set -e  # Exit immediately if a command exits with a non-zero status
+
+# Usage function
+usage() {
+    echo "Usage: $0 <vm-profile> <baseline-report>"
+    echo "Valid profiles: cannon-singlethreaded-32, cannon-multithreaded-64"
+    exit 1
+}
+
+# Validate input
+if [[ $# -ne 2 ]]; then
+    usage
+fi
+
+VM_PROFILE=$1
+BASELINE_REPORT=$2
+
+if [[ "$VM_PROFILE" != "cannon-singlethreaded-32" && "$VM_PROFILE" != "cannon-multithreaded-64" ]]; then
+    echo "Error: Invalid vm-profile '$VM_PROFILE'"
+    usage
+fi
+
+# Detect OS and architecture
+OS=$(uname -s | tr '[:upper:]' '[:lower:]')
+ARCH=$(uname -m)
+BIN_DIR="bin"
+ANALYZER_BIN="${BIN_DIR}/analyzer"
+
+# Normalize architecture naming
+case "$ARCH" in
+    x86_64) ARCH="amd64" ;;
+    aarch64) ARCH="arm64" ;;
+esac
+
+# Detect package manager
+if [[ "$OS" == "linux" ]]; then
+    if command -v apt-get &>/dev/null; then
+        PKG_MANAGER="sudo apt-get install -y"
+    elif command -v yum &>/dev/null; then
+        PKG_MANAGER="sudo yum install -y"
+    elif command -v pacman &>/dev/null; then
+        PKG_MANAGER="sudo pacman -Sy --noconfirm"
+    else
+        echo "No supported package manager found. Install llvm manually."
+        exit 1
+    fi
+elif [[ "$OS" == "darwin" ]]; then
+    PKG_MANAGER="brew install"
+else
+    echo "Unsupported OS: $OS"
+    exit 1
+fi
+
+# Detect LLVM path
+LLVM_PATH=""
+if [[ "$OS" == "darwin" ]]; then
+    if [[ "$ARCH" == "amd64" ]]; then
+        LLVM_PATH="/usr/local/opt/llvm/bin"
+    else
+        LLVM_PATH="/opt/homebrew/opt/llvm/bin"
+    fi
+elif [[ "$OS" == "linux" ]]; then
+    if command -v brew &>/dev/null; then
+        LLVM_PATH="/home/linuxbrew/.linuxbrew/opt/llvm/bin"
+    elif [[ -f "/usr/bin/llvm-objdump" ]]; then
+        LLVM_PATH="/usr/bin"
+    fi
+fi
+
+# Install llvm-objdump if missing
+if ! command -v llvm-objdump &>/dev/null; then
+    echo "llvm-objdump not found, installing..."
+    $PKG_MANAGER llvm
```

Comment:
It's probably not a good idea for users to just enter their sudo password when prompted here.  I'd suggest just printing the install commands for common package managers here and exiting with an error message that llvm-objdump is required.

We can just assume it's available if we are able to install it via mise though.

Reply:
Sounds good. Will do that.